Telegraf 1.21.0 - go panic with SNMP plugin

### Relevent telegraf.conf

```toml
Dec 17 16:38:04 hostname systemd[1]: Started The plugin-driven server agent for reporting metrics into InfluxDB.
Dec 17 16:38:04 hostname telegraf[306402]: 2021-12-17T15:38:04Z I! Starting Telegraf 1.21.1
Dec 17 16:38:04 hostname telegraf[306402]: 2021-12-17T15:38:04Z I! Loaded inputs: cpu disk internal mem net ping snmp (352x) swap
Dec 17 16:38:04 hostname telegraf[306402]: 2021-12-17T15:38:04Z I! Loaded aggregators:
Dec 17 16:38:04 hostname telegraf[306402]: 2021-12-17T15:38:04Z I! Loaded processors: converter
Dec 17 16:38:04 hostname telegraf[306402]: 2021-12-17T15:38:04Z I! Loaded outputs: influxdb_v2 (2x)
Dec 17 16:38:04 hostname telegraf[306402]: 2021-12-17T15:38:04Z I! Tags enabled: host=s
Dec 17 16:38:04 hostname telegraf[306402]: 2021-12-17T15:38:04Z I! [agent] Config: Interval:2m0s, Quiet:false, Hostname:"", Flush Interval:10s
Dec 17 16:38:06 hostname telegraf[306402]: panic: runtime error: index out of range [1] with length 1
Dec 17 16:38:06 hostname telegraf[306402]: goroutine 1 [running]:
Dec 17 16:38:06 hostname telegraf[306402]: github.com/influxdata/telegraf/internal/snmp.SnmpTranslateCall({0xc00158ca81, 0x27})
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/internal/snmp/translate.go:153 +0xbfd
Dec 17 16:38:06 hostname telegraf[306402]: github.com/influxdata/telegraf/plugins/inputs/snmp.SnmpTranslate({0xc00158ca81, 0x27})
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:877 +0x170
Dec 17 16:38:06 hostname telegraf[306402]: github.com/influxdata/telegraf/plugins/inputs/snmp.(*Field).init(0xc000766708)
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:263 +0x73
Dec 17 16:38:06 hostname telegraf[306402]: github.com/influxdata/telegraf/plugins/inputs/snmp.(*Table).Init(0xc0016ede00)
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:169 +0xd8
Dec 17 16:38:06 hostname telegraf[306402]: github.com/influxdata/telegraf/plugins/inputs/snmp.(*Snmp).Init(0xc0009f7340)
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:110 +0x127
Dec 17 16:38:06 hostname telegraf[306402]: github.com/influxdata/telegraf/models.(*RunningInput).Init(0xc000e4b758)
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/models/running_input.go:82 +0x35
Dec 17 16:38:06 hostname telegraf[306402]: github.com/influxdata/telegraf/agent.(*Agent).initPlugins(0xc000184ba8)
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/agent/agent.go:189 +0x96
Dec 17 16:38:06 hostname telegraf[306402]: github.com/influxdata/telegraf/agent.(*Agent).Run(0xc000184ba8, {0x57e5688, 0xc000acc380})
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/agent/agent.go:105 +0x185
Dec 17 16:38:06 hostname telegraf[306402]: main.runAgent({0x57e5688, 0xc000acc380}, {0x837a278, 0x0, 0x0}, {0x837a278, 0x0, 0x0})
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:312 +0xc57
Dec 17 16:38:06 hostname telegraf[306402]: main.reloadLoop({0x837a278, 0x0, 0x0}, {0x837a278, 0x0, 0x0})
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:147 +0x28a
Dec 17 16:38:06 hostname telegraf[306402]: main.run(...)
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf_posix.go:8
Dec 17 16:38:06 hostname telegraf[306402]: main.main()
Dec 17 16:38:06 hostname telegraf[306402]:         /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:485 +0xa9a
Dec 17 16:38:06 hostname systemd[1]: telegraf.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Dec 17 16:38:06 hostname systemd[1]: telegraf.service: Failed with result 'exit-code'.
Dec 17 16:38:07 hostname systemd[1]: telegraf.service: Service RestartSec=100ms expired, scheduling restart.
Dec 17 16:38:07 hostname systemd[1]: telegraf.service: Scheduled restart job, restart counter is at 26.
Dec 17 16:38:07 hostname systemd[1]: Stopped The plugin-driven server agent for reporting metrics into InfluxDB.
```


### System info

Telegraf 1.21.1

### Docker

_No response_

### Steps to reproduce

1.
After the agent restarted it restarts every 100ms because of the error "panic: runtime error: index out of range [1] with length 1"
2.
rollback to 1.21.0 does not fix the issue
rollback to 1.20.4 fixes the issue
3.
...


### Expected behavior

to work like 1.20.4

### Actual behavior

crashes

### Additional info

Not sure if I have a specific snmp MIB file that causes this, I cannot see any reference in the trace what causes it besides snmptranslate

@MyaLongmire  it still appears to panic with your PR build at least for me (using the synology mibs).


```
root@telegraf:/opt# /var/telegraf --config /etc/telegraf/telegraf.conf                                                                                                                                      
2021-12-19T10:30:59Z I! Starting Telegraf 1.22.0-9ffc58c7                                                                                                                                                   
2021-12-19T10:30:59Z I! Loaded inputs: snmp                                                                                                                                                                 
2021-12-19T10:30:59Z I! Loaded aggregators:                                                                                                                                                                 
2021-12-19T10:30:59Z I! Loaded processors:                                                                                                                                                                  
2021-12-19T10:30:59Z I! Loaded outputs: influxdb_v2                                                                                                                                                         
2021-12-19T10:30:59Z I! Tags enabled: host=ds420                                                                                                                                                            
2021-12-19T10:30:59Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"ds420", Flush Interval:10s                                                                                                     
Parse module: /usr/share/snmp/mibs/ietf/DPI20-MIB:9:4: unexpected "ibm" (expected ";")                                                                                                                      
2021-12-19T10:31:02Z W! [inputs.snmp] Could not load module at DPI20-MIB                                                                                                                                    
Parse module: /usr/share/snmp/mibs/ietf/HPR-MIB:494:30: unexpected "HprRtpCounter" (expected "}")                                                                                                           
2021-12-19T10:31:04Z W! [inputs.snmp] Could not load module at HPR-MIB                                                                                                                                      
Parse module: /usr/share/snmp/mibs/ietf/SNMPv2-PDU:73:1: unexpected "max-bindings" (expected "END")                                                                                                         
2021-12-19T10:31:08Z W! [inputs.snmp] Could not load module at SNMPv2-PDU                                                                                                                                   
Parse module: /usr/share/snmp/mibs/ietf/TCPIPX-MIB:63:12: unexpected "tcpIpxConnLocalPort" (expected "}")                                                                                                   
2021-12-19T10:31:08Z W! [inputs.snmp] Could not load module at TCPIPX-MIB                                                                                                                                   
panic: strconv.ParseUint: parsing "": invalid syntax                                                                                                                                                        
                                                                                                                                                                                                            
goroutine 1 [running]:                                                                                                                                                                                      
github.com/sleepinggenius2/gosmi/types.OidMustFromString(...)                                                                                                                                               
        /go/pkg/mod/github.com/sleepinggenius2/gosmi@v0.4.3/types/oid.go:91                                                                                                                                 
github.com/influxdata/telegraf/internal/snmp.GetIndex({0xca5084e, 0x1}, {0x10987884, 0x8})                                                                                                                  
        /go/src/github.com/influxdata/telegraf/internal/snmp/translate.go:126 +0x331                                                                                                                        
github.com/influxdata/telegraf/plugins/inputs/snmp.snmpTableCall({0x10140cd9, 0xf})                                                                                                                         
        /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:837 +0xe6                                                                                                                        
github.com/influxdata/telegraf/plugins/inputs/snmp.snmpTable({0x10140cd9, 0xf})                                                                                                                             
        /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:820 +0xeb                                                                                                                        
github.com/influxdata/telegraf/plugins/inputs/snmp.(*Table).initBuild(0xfc01860)                                                                                                                            
        /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:192 +0x44                                                                                                                        
github.com/influxdata/telegraf/plugins/inputs/snmp.(*Table).Init(0xfc01860)                                                                                                                                 
        /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:162 +0x7d                                                                                                                        
github.com/influxdata/telegraf/plugins/inputs/snmp.(*Snmp).Init(0x1003bbc0)                                                                                                                                 
        /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:110 +0x10f                                                                                                                       
github.com/influxdata/telegraf/models.(*RunningInput).Init(0x103f3e30)                                                                                                                                      
        /go/src/github.com/influxdata/telegraf/models/running_input.go:82 +0x48                                                                                                                             
github.com/influxdata/telegraf/agent.(*Agent).initPlugins(0x103df3c0)
        /go/src/github.com/influxdata/telegraf/agent/agent.go:189 +0x6c 
 github.com/influxdata/telegraf/agent.(*Agent).Run(0x103df3c0, {0xcb26974, 0x103f2300}) 
        /go/src/github.com/influxdata/telegraf/agent/agent.go:105 +0x164 main.runAgent({0xcb26974, 0x103f2300}, {0xef87ac4, Ox0, Ox0), {Oxef87ac4, Ox0, Ox0}) 
        /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:312 +0xf55 main.reloadLoop({0xef87ac4, 0x0, 0x0}, {Oxef87ac4, Ox0, 0x0}) 
        /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:147 +0x20e main.run(...) 
        /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf_posix.go:8 main. main() 
        /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:485 +0xc25
```

@MyaLongmire  You are correct looking at the issue you posted. It seem telegraf was loading the mibs from the ietf bundle. I've now removed all the RFC ones and only have the MIBs specified in my config loading and do get a different error and i'm unable to see the problem MIB.

```
2021-12-20T16:38:33Z I! Starting Telegraf 1.21.1
2021-12-20T16:38:33Z I! Using config file: /etc/telegraf/telegraf.conf
2021-12-20T16:38:33Z I! Loaded inputs: snmp
2021-12-20T16:38:33Z I! Loaded aggregators: 
2021-12-20T16:38:33Z I! Loaded processors: 
2021-12-20T16:38:33Z I! Loaded outputs: influxdb_v2
2021-12-20T16:38:33Z I! Tags enabled: host=ds420
2021-12-20T16:38:33Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"ds420", Flush Interval:10s
panic: strconv.ParseUint: parsing "": invalid syntax
goroutine 1 [running]:
github.com/sleepinggenius2/gosmi/types.OidMustFromString(...)
	/go/pkg/mod/github.com/sleepinggenius2/gosmi@v0.4.3/types/oid.go:91
github.com/influxdata/telegraf/internal/snmp.GetIndex({0x5693358, 0x1}, {0xc0011a7220, 0x8})
	/go/src/github.com/influxdata/telegraf/internal/snmp/translate.go:126 +0x3c8
github.com/influxdata/telegraf/plugins/inputs/snmp.snmpTableCall({0xc0005cb849, 0xf})
	/go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:837 +0xc5
github.com/influxdata/telegraf/plugins/inputs/snmp.snmpTable({0xc0005cb849, 0xf})
	/go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:820 +0x19d
github.com/influxdata/telegraf/plugins/inputs/snmp.(*Table).initBuild(0xc00000d860)
	/go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:192 +0x4a
github.com/influxdata/telegraf/plugins/inputs/snmp.(*Table).Init(0xc00000d860)
	/go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:162 +0x5b
github.com/influxdata/telegraf/plugins/inputs/snmp.(*Snmp).Init(0xc000623a20)
	/go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:110 +0x127
github.com/influxdata/telegraf/models.(*RunningInput).Init(0xc000ad3758)
	/go/src/github.com/influxdata/telegraf/models/running_input.go:82 +0x35
github.com/influxdata/telegraf/agent.(*Agent).initPlugins(0xc000010da8)
	/go/src/github.com/influxdata/telegraf/agent/agent.go:189 +0x96
github.com/influxdata/telegraf/agent.(*Agent).Run(0xc000010da8, {0x57e5688, 0xc00059e680})
	/go/src/github.com/influxdata/telegraf/agent/agent.go:105 +0x185
main.runAgent({0x57e5688, 0xc00059e680}, {0x837a278, 0x0, 0x0}, {0x837a278, 0x0, 0x0})
	/go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:312 +0xc57
main.reloadLoop({0x837a278, 0x0, 0x0}, {0x837a278, 0x0, 0x0})
	/go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:147 +0x28a
main.run(...)
	/go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf_posix.go:8
main.main()
	/go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:485 +0xa9a
```

Attached is the config and the mibs
[config.zip](https://github.com/influxdata/telegraf/files/7746739/config.zip)
.

--- a/internal/snmp/translate.go
+++ b/internal/snmp/translate.go
@@ -148,26 +148,25 @@ func SnmpTranslateCall(oid string) (mibName string, oidNum string, oidText strin
 	if strings.ContainsAny(oid, "::") {
 		// split given oid
 		// for example RFC1213-MIB::sysUpTime.0
-		s := strings.Split(oid, "::")
+		s := strings.SplitN(oid, "::", 2)
 		// node becomes sysUpTime.0
-		if s[1] != "" {
-			node := s[1]
-			if strings.ContainsAny(node, ".") {
-				s = strings.Split(node, ".")
-				// node becomes sysUpTime
-				node = s[0]
-				end = "." + s[1]
-			}
-
-			out, err = gosmi.GetNode(node)
-			if err != nil {
-				return oid, oid, oid, oid, err
-			}
-
-			oidNum = "." + out.RenderNumeric() + end
-		} else {
-			return "", oid, oid, oid, fmt.Errorf("can not parse %v\n", oid)
+		if s[1] == "" {
+			return "", oid, oid, oid, fmt.Errorf("cannot parse %v\n", oid)
 		}
+		node := s[1]
+		if strings.ContainsAny(node, ".") {
+			s = strings.SplitN(node, ".", 2)
+			// node becomes sysUpTime
+			node = s[0]
+			end = "." + s[1]
+		}
+
+		out, err = gosmi.GetNode(node)
+		if err != nil {
+			return oid, oid, oid, oid, err
+		}
+
+		oidNum = "." + out.RenderNumeric() + end
 	} else if strings.ContainsAny(oid, "abcdefghijklnmopqrstuvwxyz") {
 		//handle mixed oid ex. .iso.2.3
 		s := strings.Split(oid, ".")

--- a/internal/snmp/translate.go
+++ b/internal/snmp/translate.go
@@ -150,20 +150,24 @@ func SnmpTranslateCall(oid string) (mibName string, oidNum string, oidText strin
 		// for example RFC1213-MIB::sysUpTime.0
 		s := strings.Split(oid, "::")
 		// node becomes sysUpTime.0
-		node := s[1]
-		if strings.ContainsAny(node, ".") {
-			s = strings.Split(node, ".")
-			// node becomes sysUpTime
-			node = s[0]
-			end = "." + s[1]
-		}
+		if s[1] != "" {
+			node := s[1]
+			if strings.ContainsAny(node, ".") {
+				s = strings.Split(node, ".")
+				// node becomes sysUpTime
+				node = s[0]
+				end = "." + s[1]
+			}
 
-		out, err = gosmi.GetNode(node)
-		if err != nil {
-			return oid, oid, oid, oid, err
-		}
+			out, err = gosmi.GetNode(node)
+			if err != nil {
+				return oid, oid, oid, oid, err
+			}
 
-		oidNum = "." + out.RenderNumeric() + end
+			oidNum = "." + out.RenderNumeric() + end
+		} else {
+			return "", oid, oid, oid, fmt.Errorf("can not parse %v\n", oid)
+		}
 	} else if strings.ContainsAny(oid, "abcdefghijklnmopqrstuvwxyz") {
 		//handle mixed oid ex. .iso.2.3
 		s := strings.Split(oid, ".")

--- a/plugins/inputs/snmp/snmp_test.go
+++ b/plugins/inputs/snmp/snmp_test.go
@@ -1304,3 +1304,14 @@ func BenchmarkMibLoading(b *testing.B) {
 		require.NoError(b, err)
 	}
 }
+
+func TestCanNotParse(t *testing.T) {
+	s := &Snmp{
+		Fields: []Field{
+			{Oid: "RFC1213-MIB::"},
+		},
+	}
+
+	err := s.Init()
+	require.Error(t, err)
+}

